editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -533,12 +559,24 @@ void SpriteFramesEditor::_prepare_sprite_sheet(const String &p_file) {
 		split_sheet_sep_y->set_value(0);
 		split_sheet_offset_x->set_value(0);
 		split_sheet_offset_y->set_value(0);
-		updating_split_settings = false;
-
-		// Reset zoom.
-		_sheet_zoom_reset();
+	} else {
+		// If there's a last_texture then load the data.
+		split_sheet_h->set_value(split_data["hor"]);
```

Comment:
Is this data really guaranteed to be available at this point? Does having a `last_texture` guarantee that all elements are filled in?

Reply:
My question here is: can you guarantee that the dictionary is valid and has all the entries? Otherwise it is going to crash here AFAIK

Reply:
as I told you, no problem in my testing, if you're reviewing you can test this by yourself!

Reply:
You are fetching a default value of `Dictionary()`, please tell me how you know that this default value will never be used here, I'm trying to help here, by sharing my own experience that just because you tested it in one way doesn't mean it won't ever go wrong, that's part of writing code 🙃

Reply:
So while I can't get it to crash (the getter for dictionary doesn't crash with a non-const reference) it does break if you do (you will have to wipe the entries from your `project_metadata.cfg` in order to have a clean slate):
* Open a sprite sheet
* Without changing any of the settings, open a new one with the "Change Spritesheet" button
* See how the entries are all invalid

This needs to be checked properly, that the dictionary is valid when loading here

Reply:
thanks for being clear, you were right, the error is confirmed, as you requested here's a robust solution, can you please test and give me a review, best regards...

changes are in lines 539-542

Reply:
This doesn't take all my concerns into account, but will test when I'm able

Reply:
yeah I realised that, sorry I'm not too experienced, but I didnt understood well how does `dominant_param` works exacly, I know that size X and Y is calculated and doesnt need to be stored, but if you could explain me better how this var works in the godot chat I would love to make the necessary changes!

a problem that I think that could happen is:

* update a texture size via image editor, and reopens in godot, it will still thinks that the size is the same, thus loading old size X and Y values

Reply:
So `dominant_param` is set by changing either:
* The H/V, in which case it makes the size calculated
* The size, in which case the H/V is calculated

The code how it works is here:
https://github.com/godotengine/godot/blob/5cfa9a0d7b0d777d550662792e28dcfd6e84d459/editor/plugins/sprite_frames_editor_plugin.cpp#L445-L479

So I'd say that you should:
* Save the `dominant_param`
* Save either `hor/vert` or `size_x/size_y` depending on `dominant_param`
* Save the offset/spacing always